engine/src/cubos/engine/tools/palette_editor/plugin.cpp

```diff
@@ -0,0 +1,79 @@
+#include <cubos/core/data/old/debug_serializer.hpp>
+#include <cubos/core/log.hpp>
+
+#include <cubos/engine/assets/asset.hpp>
+#include <cubos/engine/imgui/plugin.hpp>
+#include <cubos/engine/renderer/plugin.hpp>
+#include <cubos/engine/tools/asset_explorer/plugin.hpp>
+#include <cubos/engine/tools/palette_editor/plugin.hpp>
+#include <cubos/engine/voxels/palette.hpp>
+#include <cubos/engine/voxels/plugin.hpp>
+
+#include "imgui.h"
+
+using cubos::core::data::old::Debug;
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+using namespace cubos::engine;
+
+struct SelectedPaletteInfo
+{
+    VoxelPalette palette;
+};
+
+static void checkAssetEventSystem(cubos::core::ecs::EventReader<tools::AssetSelectedEvent> reader, Read<Assets> assets,
+                                  Write<SelectedPaletteInfo> selectedPalette)
```

Comment:
```suggestion
static void checkAssetEventSystem(EventReader<tools::AssetSelectedEvent> reader, Read<Assets> assets,
                                  Write<SelectedPaletteInfo> selectedPalette)
```
and add an `using cubos::core::ecs::EventReader` above